nuxeo-services/nuxeo-platform-query-api/src/main/java/org/nuxeo/ecm/platform/query/nxql/NXQLQueryBuilder.java

```diff
@@ -717,4 +722,17 @@ public static String appendClause(String query, String clause) {
         return query + " AND " + clause;
     }
 
+    private static String buildPattern(String pattern, String key, String replacement) {
+        int index = pattern.indexOf(key);
+        while (index >= 0) {
+            // All keys not prefixed by a letter or a digit has to be replaced, because
+            // It could be part of a schema name
+            if (!Character.isLetterOrDigit(pattern.charAt(index - 1))) {
+                pattern = pattern.substring(0, index) + pattern.substring(index).replaceFirst(key, replacement);
+            }
+            index = pattern.indexOf(key, index + 1);
+        }
+        return pattern;
+    }
+
```

Comment:
I think it's worth making this a `public` method and adding a few unit tests for just this that make it clear what it does. At least show in tests the basic cases, zero, one and two replacements. It'd be worth testing also the case where in `queryParams` you have `myVar` and `myVariable` and show that they are not confused. The current code you wrote (like the previous one) still has this issue as well.